src/status_im/contexts/wallet/wallet_connect/events.cljs

```diff
@@ -50,13 +51,28 @@
               {:error error
                :event :wallet-connect/on-init-fail})))
 
+(defn- accounts-without-watched-with-customization-color
+  [all-accounts]
+  (->> all-accounts
+       (remove :watch-only?)
+       wallet-utils/accounts-with-customization-color))
+
 (rf/reg-event-fx
  :wallet-connect/on-session-proposal
  (fn [{:keys [db]} [proposal]]
-   (log/info "Received Wallet Connect session proposal: " {:id (:id proposal)})
-   {:db (assoc db :wallet-connect/current-proposal proposal)
-    :fx [[:dispatch
-          [:open-modal :screen/wallet.wallet-connect-session-proposal]]]}))
+   (let [all-accounts (get-in db [:wallet :accounts])
+         accounts     (accounts-without-watched-with-customization-color all-accounts)]
+     (log/info "Received Wallet Connect session proposal: " {:id (:id proposal)})
+     {:db (assoc db
+                 :wallet-connect/current-proposal         proposal
+                 ;; NOTE: for now using the first account, but should be using the account selected
+                 ;; by the user on the connection screen. The default would depend on where the
+                 ;; connection started from:
+                 ;; - global scanner -> first account in list
+                 ;; - wallet account dapps -> account that is selected
+                 :wallet-connect/current-proposal-account (first accounts))
+      :fx [[:dispatch
+            [:open-modal :screen/wallet.wallet-connect-session-proposal]]]})))
```

Comment:
Although account switcher works, this note is still needed as we have not built the `wallet account dapps` case yet.